trl/trainer/dpo_trainer.py

```diff
@@ -1278,16 +1457,21 @@ def get_batch_loss_metrics(
 
         model_output = self.concatenated_forward(model, batch)
```

Comment:
can we do this instead? Probably simpler

```python
if self.use_liger_loss:
    losses, chosen_rewards, rejected_rewards = self._compute_loss_liger(model, batch)
else:
    model_output = self.concatenated_forward(model, batch)

    # if ref_chosen_logps and ref_rejected_logps in batch use them, otherwise use the reference model
    if "ref_chosen_logps" in batch and "ref_rejected_logps" in batch:
        ref_chosen_logps = batch["ref_chosen_logps"]
        ref_rejected_logps = batch["ref_rejected_logps"]
    else:
        ref_chosen_logps, ref_rejected_logps = self.compute_ref_log_probs(batch)

    losses, chosen_rewards, rejected_rewards = self.dpo_loss(
        model_output["chosen_logps"], model_output["rejected_logps"], ref_chosen_logps, ref_rejected_logps
    )
```